test/e2e/test_pullsubscription.go

```diff
@@ -34,20 +34,21 @@ import (
 )
 
 // SmokePullSubscriptionTestImpl tests we can create a pull subscription to ready state.
-func SmokePullSubscriptionTestImpl(t *testing.T) {
+func SmokePullSubscriptionTestImpl(t *testing.T, authConfig lib.AuthConfig) {
 	topic, deleteTopic := lib.MakeTopicOrDie(t)
 	defer deleteTopic()
 
 	psName := topic + "-sub"
 	svcName := "event-display"
 
-	client := lib.Setup(t, true)
+	client := lib.Setup(t, true, authConfig.WorkloadIdentityEnabled)
 	defer lib.TearDown(client)
 
 	// Create PullSubscription.
 	pullsubscription := kngcptesting.NewPullSubscription(psName, client.Namespace,
 		kngcptesting.WithPullSubscriptionSpec(v1alpha1.PullSubscriptionSpec{
-			Topic: topic,
+			Topic:          topic,
+			ServiceAccount: authConfig.PubsubServiceAccount,
```

Comment:
```suggestion
			IdentitySpec: duckv1alpha1.IdentitySpec{
				ServiceAccount: authConfig.PubsubServiceAccount,
			},
```

Something like this. Also needed on line 81.